modules/core/arrow-core-data/src/test/kotlin/arrow/typeclasses/MonoidTest.kt

```diff
@@ -1,19 +1,23 @@
 package arrow.typeclasses
 
+import arrow.Kind
 import arrow.core.extensions.monoid
 import arrow.core.extensions.monoid.invariant.invariant
 import arrow.test.UnitSpec
 import arrow.test.laws.InvariantLaws
+import io.kotlintest.properties.Gen
 
 class MonoidTest : UnitSpec() {
 
-    val EQ: Eq<MonoidOf<Int>> = Eq.invoke { a, b ->
-      a.fix().run { 3.combine(1) } == b.fix().run { 3.combine(1) }
-    }
+  val EQ: Eq<MonoidOf<Int>> = Eq.invoke { a, b ->
+    a.fix().run { 3.combine(1) } == b.fix().run { 3.combine(1) }
+  }
 
-    init {
-        testLaws(
-            InvariantLaws.laws(Monoid.invariant<Int>(), { Int.monoid() }, EQ)
-        )
-    }
+  val GEN = Gen.constant(Int.monoid()) as Gen<Kind<ForMonoid, Int>>
```

Comment:
There is also a `GenK` and `EqK` instance here ^^

Reply:
i dont understand what you mean here.
also the current state of the test does not seem to make much sense to me.
all it seems to do is combine empty structures? see also my comment in the test.

Reply:
Well yes, both of those tests were pointless. Checking if a monoid instance equals another means checking the result of `empty` and checking `combine`. But `combine` is a function and checking function equality is quite hard if not impossible. Just keep it as is for now, there is basically no better way. As I said, the previous one was just as bad